[NodeJS Security Cheat Sheet] Update various helmet middlewares

maxime-bc opened the pull request:
This PR update various helmet middlewares found in the **NodeJS Security Cheat Sheet**.

- Replace `express-session` cookie `key` attribute with `name`. Although the `key` attribute still works, [it was replaced by `name`](https://github.com/expressjs/session/blob/master/HISTORY.md#110--2014-05-12) and only the later is referenced in the documentation,
- Replace `helmet.xframe()` with `helmet.frameguard()` : [the xframe middleware is now called frameguard](https://github.com/helmetjs/helmet/blob/main/CHANGELOG.md#050---2014-10-28) (xframe still works). Also, updated the example as [frameguard is now set to SAMEORIGIN by default](https://github.com/helmetjs/helmet/blob/main/CHANGELOG.md#600---2022-08-26), and ALLOW-FROM is no more supported,
- Update the X-XSS-Protection section as it is now disabled by default by `helmet`. See : https://github.com/helmetjs/helmet/blob/main/CHANGELOG.md#changed-12,
- Replace `helmet.noCache()` with the `nocache` package. See : https://github.com/helmetjs/helmet/blob/main/CHANGELOG.md#removed-4,
- Add information and example about the top level `helmet()` function,
- Some small grammatical fixes.

Also, I saw that the section [Cross_Site_Scripting_Prevention_Cheat_Sheet.md#x-xss-protection-header](https://github.com/OWASP/CheatSheetSeries/blob/9d5dc5b52c07eb4af4dcfadd8803f06c8d55918b/cheatsheets/Nodejs_Security_Cheat_Sheet.md?plain=1#L451) no longer exists in the `Cross_Site_Scripting_Prevention_Cheat_Sheet.md` cheatsheet. Should the link be deleted or should it point to another section ?

szh commented on cheatsheets/Nodejs_Security_Cheat_Sheet.md:
```diff
@@ -424,33 +424,51 @@ Generally, session information is sent using cookies in web applications. Howeve
 
-There are several different [HTTP security headers](https://owasp.org/www-project-secure-headers/) that can help you prevent some common attack vectors. These are listed below:
+There are several [HTTP security headers](https://owasp.org/www-project-secure-headers/) that can help you prevent some common attack vectors.
+The [helmet](https://www.npmjs.com/package/helmet) package can help to set those headers :
```
```suggestion
The [helmet](https://www.npmjs.com/package/helmet) package can help to set those headers:
```

szh commented on cheatsheets/Nodejs_Security_Cheat_Sheet.md:
````diff
@@ -473,10 +491,12 @@ app.use(csp({
 ```
 
-- **[Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) and [Pragma](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Pragma):** Cache-Control header can be used to prevent browsers from caching the given responses. This should be done for pages that contain sensitive information about either the user or the application. However, disabling caching for ...
+- **[Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) and [Pragma](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Pragma):** Cache-Control header can be used to prevent browsers from caching the given responses. This should be done for pages that contain sensitive information about either the user or the application. However, disabling caching for ...
````
```suggestion
- **[Cache-Control](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) and [Pragma](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Pragma):** Cache-Control header can be used to prevent browsers from caching the given responses. This should be done for pages that contain sensitive information about either the user or the application. However, disabling caching for pages that do not contain sensitive information may seriously affect the performance of the application. Therefore, caching should only be disabled for pages that return sensitive information. Appropriate caching controls and headers can be set easily using the [nocache](https://www.npmjs.com/package/nocache) package:
```

szh reviewed (commented):
Thank you for the detailed PR. I really appreciate all the links to specific changes.
I left a couple of nits but otherwise it looks good.
In terms of the X-XSS-Protection header, MDN doesn't recommend it which is probably why it's no longer in the XSS cheat sheet. I think we can just remove it from this one as well.

maxime-bc commented:
Hello @szh, @jmanico, 

I have done the following changes :

- Removed spaces before colons,
- Removed the dead link to the **XSS Prevention Cheat Sheet**,
- Added information about the **X-XSS-Protection** header and why it should be forcibly disabled. I also moved the **CSP** section just below the **X-XSS-Protection** as they are linked,
- Updated the **CSP** header example : there is no need to use the `helmet-csp` module as the **CSP** middleware is already bundled with `helmet`.

szh reviewed (approved):
LGTM! Thank you so much!
